### DISCUSSION TITLE
[2023-09-11] GUI Team Demo Day

### DISCUSSION THREAD

**Discussion by farmaazon** [enso-org/enso#7792] [Weekly Demo Day]
The @jdunkerley's filtering and scoring algorithm is implemented. Tested on the mock data. In the screencast below, the scores are in parentheses (or they are undefined if not matched at all).

https://github.com/enso-org/enso/assets/3919101/14701d1e-b787-46c9-8234-461e7844e3c4

Some work is still needed to finish it, most prominently the ordering of elements of the same score and styling.

**Comment by wdanilo** [comment]
I was working on the new website. Notable changes:

- It is not a draft anymore – the screen size handling is implemented and it was a total pain in the ass. I wanted to not refresh DOM during screen changes, but with many animations and moving elements it was hard. Also, the fact that SVG and HTML use different coordinate systems with different rules is ugly and I needed to do a lot of adjustments to that. Anyway, the screen handling is done and works no matter the height of the screen. The width is not yet adjusted, but it will be easy.
- I worked on copy (text on website) to be short, informative, and partitioned in the right way.
- I added additional animations when scrolling the web page.
- I added top-bar which shows when the website is scrolled and hides if you scroll up.

The below video shows what's ready. There are multiple elements in the WIP state that are not visible here:


https://github.com/enso-org/enso/assets/1623053/1dcd9709-0e05-4067-aa58-15229f712c11

**Comment by vitvakatu** [comment]
Last week I:
- Made all nodes have the same color: https://github.com/enso-org/enso/pull/7764
- Fixed a minor issue with pasting nodes: https://github.com/enso-org/enso/pull/7768
- Fixed the issue with broken fullscreen visualization focus: https://github.com/enso-org/enso/pull/7782
- Investigating https://github.com/enso-org/enso/issues/7457 – should be soon.
- Almost lost my mind working on https://github.com/enso-org/enso/pull/7779, but it's finally on review! There is a ~~small issue~~ possible improvement to undo-redo behavior, which should be ready today.


https://github.com/enso-org/enso/assets/6566674/2459feb1-4c91-487e-aac0-a188372eab29

**Comment by kazcw** [comment]
- Fixed #7691 by refactoring a parser component to scale to large inputs; this also improves performance for typical inputs [PR will be ready for review today.]
- Designed a minimal-allocation deserialization API to get Rust ASTs into TS efficiently; started implementation (#7701). The mechanism behind this API will also support tree-diffing, to enable efficient incremental parsing.

**Comment by somebody1234** [comment]
I was working on:
- Top bar - addressing reviews: #7726
![image](https://github.com/enso-org/enso/assets/4046547/4a26d10b-239b-4da8-9e50-4d26512cab00)
- Basic widgets (checkbox, slider, dropdown): #7728
- Visualizations: #7773
  - Porting existing visualizations to Vue
  - Deciding on an appropriate API for Vue visualizations
  - Implementing on-the-fly transpilation (at runtime, purely in the browser) from Vue Single File Components to plain JavaScript
  - This is almost ready, however there are few issues with the visualizations, and there are still one or two visualizations left to port over.


https://github.com/enso-org/enso/assets/4046547/4d153286-54e3-4f57-8740-885725437849

**Comment by mwu-tow** [comment]
Last week I was working on:
* Ongoing CI maintenance;
* Support for commercial AG Grid (#7720);
* Adding support for the new GUI to the build scripts;
* Packaging the new GUI in the Electron.

![image](https://github.com/enso-org/enso/assets/1548407/12f48201-6dc9-4e2f-b248-1ad6b563f1aa)

**Comment by Frizi** [comment]
- added basic version of enso rust parser to use in new gui
- integrated new gui with dashboard, allowing us to load any projects and receive URL to language server
- working on synchronization between project files in language server and shared CRDT data. Currently have the connection set up and I'm implementing RPC methods and proper reaction to changes reported by both parties (LS/CRDT).

https://github.com/enso-org/enso/assets/919491/425f5d4b-5bbe-4484-819a-d17c59443004

**Comment by hubertp** [comment]
Relevant GUI changes:
- expression updates for binary operators #7659
